helm/flux-app/templates/install.yaml

```diff
@@ -687,6 +687,11 @@ spec:
       annotations:
         prometheus.io/port: "8080"
         prometheus.io/scrape: "true"
+{{- if (.Values.kustomizeController.podTemplate.annotations) }}
```

Comment:
It works of course, but I think we need to figure out of how to do it a bit differently. The reason being we do not manage this file directly, but instead generate it from the upstream while adding some of our stuff on the way with `kustomize`, see the `hack` directory. Adding conditional block like this with `kustomize` will be hard at least I think, since `kustomize` expects YAML format, which the template blocks contradict. 

Instead I think we can move the whole annotations map to `_helpers.tpl`, along with what you've added I think, like this:

```text
{{- define "podTemplateAnnotations.kustomizeController" -}}
{{- printf "prometheus.io/port: \"8080\"" | nindent 8 }}
{{- printf "prometheus.io/scrape: \"true\"" | nindent 8 }}
{{- if (.Values.kustomizeController.podTemplate.annotations) }}
{{- range $k, $v := .Values.kustomizeController.podTemplate.annotations }}
{{- $k | nindent 8 }}: {{ $v }}
{{- end }}
{{- end }}
{{- end -}}
```

Then we could change this part to just:

```yaml
...
      annoattions: {{ include "podTemplateAnnotations.kustomizeController" . }}
```

And then [here](https://github.com/giantswarm/flux-app/blob/master/hack/patches/kustomize-controller-deploy-patch.yaml) in the hack directory we could replace the value of annotations, like:

```yaml
- op: replace
  path: /spec/template/metadata/annotations
  value: '{{ include "podTemplateAnnotations.kustomizeController" . }}'
```

This way we won't loose the functionality on the next generation. The only thing I don't like in turn in this solution is that we need to move Prometheus labels to the `_helpers.tpl`, and we do not know which labels may be added to the file in the future by the upstream.

Reply:
I think that is just something we have to track - I am not aware of any clean solution to merging upstream labels when `hack`ing a chart in this way.